src/ctia/flows/crud.clj

```diff
@@ -296,6 +300,49 @@
     :delete (first results)
     :update (first entities)))
 
+(defn recast [orig-coll new-coll]
+  (cond
+    (vector? orig-coll) (vec new-coll)
+    (set? orig-coll) (set new-coll)
+    :else new-coll))
+
+(defn add-colls [& args]
+  (let [new-coll
+        (->> args
+             (map #(or % []))
+             (apply into))]
+    (recast (first args) new-coll)))
+
+(defn remove-colls [& args]
+  (let [new-coll
+        (reduce
+         (fn [a b]
+           (remove (or (set b) #{})
+                   (or a []))) args)]
+    (recast (first args) new-coll)))
+
+(defn replace-colls [& args]
+  (let [new-coll (last args)]
+    (recast (first args) new-coll)))
+
+(s/defn patch-entities :- FlowMap
+  [{:keys [prev-entity
+           partial-entity
+           patch-operation]
+    :as fm} :- FlowMap]
+
+  (let [patch-fn (case patch-operation
+                   :add add-colls
+                   :remove remove-colls
+                   :replace replace-colls
+                   replace-colls)
+        entity (-> (deep-merge-with patch-fn
+                                    prev-entity
+                                    partial-entity)
+                   un-store
+                   (dissoc :id))]
+    (assoc fm :entities [entity])))
+
```

Comment:
All those functions are really "data manipulation" about things without types or form.
Its hard to follow. Also I couldn't find any test showing how to use them and what to expect from them individually.

Would you mind to add a docstring with example and if possible unit test for each of these functions with simple examples? Typically for `add-colls`, `remove-colls` and `replace-colls`.